sympy/core/numbers.py

```diff
@@ -1388,23 +1388,10 @@ def _Frel(self, other, op):
                 else:
                     # self and other have the same sign and other
```

Comment:
It is probably not necessary to look at the signs as `mpf_cmp` also implements the shortcut of different signs.

Reply:
The `is_integer` does not look like it will meet the expectations of the codebase.